assets/src/edit-story/components/dropDown/index.js

```diff
@@ -0,0 +1,314 @@
+  }
+
+  &:focus {
+    background-color: ${({ theme }) => theme.colors.bg.v11};
+    outline: none;
+  }
+`;
+
+const DropDownItemSpan = styled.a``;
+
+function DropDown({ options, value, onChange, disabled }) {
+  DropDown.selectRef = React.createRef();
+  DropDown.arrayOfOptionsRefs = [];
+
+  const [openOptions, setOpenOptions] = useState(false);
+  const [focusedOption, setFocusedOption] = useState(undefined);
+  const [searchValue, setSearchValue] = useState('');
+  const [activeItem, setActiveItem] = useState(
+    options.find((item) => item.value === value)
+  );
+
+  useEffect(() => {
+    setActiveItem(options.find((item) => item.value === value));
+  }, [value, options]);
+
+  DropDown.handleClickOutside = () => setOpenOptions(false);
+
+  const clearOptionsRefs = () => {
+    DropDown.arrayOfOptionsRefs = [];
+  };
+
+  const handleOpenOptions = (event) => {
+    switch (event.type) {
+      case 'click':
+        setOpenOptions(!openOptions);
+        setFocusedOption(document.activeElement.id);
+        break;
+      case 'keydown':
+        if (event.key === 'Enter' || event.key === ' ') {
+          _handleOpenOptions(event);
+        }
+        break;
+      default:
+    }
+  };
+
+  const _handleOpenOptions = useCallback(() => {
+    setOpenOptions(!openOptions);
+  }, [openOptions]);
+
+  const handleCurrentValue = useCallback(
+    (option) => {
+      if (onChange) {
+        onChange(option);
+      }
+    },
+    [onChange]
+  );
+
+  const handleSelectElement = (event, index) => {
+    event.preventDefault();
+    if (index >= 0 && index < DropDown.arrayOfOptionsRefs.length) {
+      DropDown.arrayOfOptionsRefs[index].focus();
+      setFocusedOption(document.activeElement.id);
+    }
+  };
+
+  const handleOptionsEvents = (option, index, event) => {
+    switch (event.type) {
+      case 'click':
+        handleCurrentValue(option);
+        setOpenOptions(!openOptions);
+        break;
+      case 'keydown':
+        if (event.key === 'Enter' || event.key === ' ') {
+          handleCurrentValue(option);
+          setOpenOptions(!openOptions);
+          DropDown.selectRef.current.focus();
+        } else if (event.key === 'ArrowUp') {
+          handleSelectElement(event, index - 1);
+        } else if (event.key === 'ArrowDown') {
+          handleSelectElement(event, index + 1);
+        } else if (event.key === 'Escape') {
+          event.preventDefault();
+          setOpenOptions(!openOptions);
+          DropDown.selectRef.current.focus();
+        } else {
+          const searchTerm = searchValue + String.fromCharCode(event.keyCode);
+          setSearchValue(searchTerm);
+          clearSearchValue();
+        }
+        break;
+      default:
+    }
+  };
+
+  const clearSearchValue = useCallback(
+    debounce(800, () => {
+      setSearchValue('');
+    })
+  );
+
+  useEffect(() => {
+    if (searchValue !== '') {
+      const searchIndex = options.findIndex((item) =>
```

Comment:
This search find a string inside the option. Should it search the start of the string?

Reply:
That makes sense for me. Thank you